### PR TITLE
[AR-132] Add transaction types on to provenance activity records

### PR DESCRIPTION
**Pull request by probonas**
Two notes here:
1. Transactions of type _Withdrawn_ do not create any provenance records in the sales pipeline
2. Transactions of type _Exchanged,Expensed,Destroyed_ do not create any provenance records in the knoedler pipeline

### CODE REVIEW

File: pipeline/projects/knoedler/__init__.py [hunk 1]
```diff
@@ -723,10 +723,7 @@ def modifiers(self, a:dict, key:str):
 	def _empty_tx(self, data, incoming=False, purpose=None):
 		tx_uri = self.helper.transaction_uri_for_record(data, incoming)
 		tx_type = data.get('book_record', {}).get('transaction', 'Sold')
-		if purpose == 'returning':
-			tx = vocab.make_multitype_obj(vocab.SaleAsReturn, vocab.ProvenanceEntry, ident=tx_uri)
-		else:
-			tx = vocab.ProvenanceEntry(ident=tx_uri)
```
Comment (kasei) [inline pipeline/projects/knoedler/__init__.py:729]: This change seems outside the scope of the the description of AR-132. Can you please add context for why this is being done (was this based on a private conversation?), or revert it?

File: pipeline/projects/knoedler/__init__.py [hunk 2]
```diff
@@ -1305,15 +1320,16 @@ class ModelReturn(ModelSale):
 	helper = Option(required=True)
 	make_la_person = Service('make_la_person')
 	buy_sell_modifiers = Service('buy_sell_modifiers')
+	transaction_classification = Service('transaction_classification')
 
-	def __call__(self, data:dict, make_la_person, buy_sell_modifiers):
+	def __call__(self, data:dict, make_la_person, buy_sell_modifiers,transaction_classification):
```
Comment (kasei) [inline pipeline/projects/knoedler/__init__.py:1325]: This is very minor, but there are several places in this PR where use (lack) of whitespace differs from surrounding code. If possible, could this be standardized?

Comment (kasei) [inline pipeline/projects/knoedler/__init__.py:1325]: This is very minor, but there are several places in this PR where use (lack) of whitespace differs from surrounding code. If possible, could this be standardized?

File: tests/test_knoedler_issue_ar132.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+from cromulent import vocab
+
+vocab.add_attribute_assignment_check()
+
+class PIRModelingTest_AR132(TestKnoedlerPipelineOutput):
+    def test_modeling_ar132(self):
+        '''
+        AR-132: Add transaction types on to Knoedler provenance activity records
+        '''
+        output = self.run_pipeline('ar132')
```
Comment (kasei) [inline tests/test_knoedler_issue_ar132.py:14]: The corresponding test data files seem to be missing from this PR.

Comment (kasei) [inline tests/test_knoedler_issue_ar132.py:14]: The corresponding test data files seem to be missing from this PR.

**Review by kasei — CHANGES_REQUESTED** [review]

File: pipeline/projects/knoedler/__init__.py (hunk 2 above)
Comment (probonas) [reply pipeline/projects/knoedler/__init__.py:1325]: Do you use a formatter like black or you fix those things manually?

File: tests/test_sales_issue_ar132.py [hunk 4]
```diff
@@ -17,13 +17,13 @@ def test_modeling_ar132(self):
...
-            'tag:getty.edu,2019:digital:pipeline:REPLACE-WITH-UUID:sales#PROV,B-183,1810-10-26,0035' : 'Bought In'
+            'tag:getty.edu,2019:digital:pipeline:REPLACE-WITH-UUID:sales#PROV,B-A138,1774-05-30,0050' : 'Purchase',
+            'tag:getty.edu,2019:digital:pipeline:REPLACE-WITH-UUID:sales#PROV-MULTI,B-A111,1769-07-31,13,14' : 'Bought In'
         }
 
         expected_id = {
-            'tag:getty.edu,2019:digital:pipeline:REPLACE-WITH-UUID:sales#PROV,B-183,1810-10-26,0005' : 'http://vocab.getty.edu/aat/300445698', # passed has no label
-            'tag:getty.edu,2019:digital:pipeline:REPLACE-WITH-UUID:sales#PROV,B-183,1810-10-26,0015' : 'https://data.getty.edu/local/thesaurus/unknown', #unknown has no label
+            'tag:getty.edu,2019:digital:pipeline:REPLACE-WITH-UUID:sales#PROV,Br-4716,1837-03-11,0087' : 'http://vocab.getty.edu/aat/300445698', # passed has no label
+            'tag:getty.edu,2019:digital:pipeline:REPLACE-WITH-UUID:sales#PROV,B-A136,1773-07-20,0090' : 'https://data.getty.edu/local/thesaurus/unknown', #unknown has no label
```
Comment (kasei) [inline tests/test_sales_issue_ar132.py:26]: I'll let George make the decision on this, but I think we should try to ensure all the types have a label (making the use of `expected_id` here unnecessary).

**Review by kasei — APPROVED** [review]
This looks good. I'll pass it along to George.
